web.py

```diff
@@ -41,82 +43,21 @@ def sh_profile_info():
             logger.warn(e)
     return make_response("", response_code)
 
-@app.route('/signing-flows/<signflow_id>/pieces')
-def pieces_get(signflow_id):
-    signflow_uri = get_by_uuid(signflow_id)
-    records = signing_flow.get_pieces(signflow_uri)
-
-    data = [{
-        "type": "pieces",
-        "id": r["id"],
-        "attributes": {
-            "uri": r["uri"]
-        }
-    } for r in records]
-    res = make_response({ "data": data }, 200)
-    res.headers["Content-Type"] = "application/vnd.api+json"
-    return res
 
-
-@app.route('/signing-flows/<signflow_id>/upload-to-signinghub', methods=['POST'])
+@app.route('/signing-flows/upload-to-signinghub', methods=['POST'])
 @jsonapi.header_required
 @signinghub_session_required  # provides g.sh_session
-def prepare_post(signflow_id):
-    signflow_uri = get_by_uuid(signflow_id)
-    pieces = signing_flow.get_pieces(signflow_uri)
-    if any(piece["sh_document_id"] for piece in pieces):
-        raise Exception("Signingflow has already been uploaded previously.")
-
-    piece_uris = [piece["uri"] for piece in pieces]
+def prepare_post():
+    body = request.get_json(force=True)
 
-    prepare_signing_flow.prepare_signing_flow(g.sh_session, signflow_uri, piece_uris)
+    sign_flow_ids = [entry["id"] for entry in body["data"]]
 
-    res = make_response("", 204)
-    res.headers["Content-Type"] = "application/vnd.api+json"
-    return res
+    query = construct_get_signing_flows_by_uuids(sign_flow_ids)
+    sign_flows = to_recs(agent_query(query))
```

Comment:
This could be a regular query, no?

Reply:
As in, querying 1-per-1 using `get_by_uuid` and separate queries per sign flow to get the decision activity, report, piece? Yes. Or do you mean something else?

Reply:
Oh, no, I meant `agent_query` provides more privileges than are actually needed. `query` could be used instead`?

Reply:
Yep, completely right. Just an automatism of using `agent_query` since it was already imported in the file.